<div class="alert alert-block alert-info"><b><h1>Univariate </h1></b></div>

<div class="alert alert-block alert-info"><b><h3>Data Preparation</h3></b></div>

In [6]:
df_train = pd.read_csv("Airline_Passenger_Satisfaction_Train_Cleaned.csv")
df_test = pd.read_csv("Airline_Passenger_Satisfaction_Test_Cleaned.csv")

def conv_cat(data):
    data['CustomerType'] = data['CustomerType'].map({'Loyal Customer': 1, 'disloyal Customer': 0})
    data['TypeOfTravel'] = data['TypeOfTravel'].map({'Personal Travel':0, 'Business travel':1})
    data['Class'] = data['Class'].map({'Business':2, 'Eco Plus':1, 'Eco':0})
    data['satisfaction'] = data['satisfaction'].map({'neutral or dissatisfied':0, 'satisfied':1})
    data = pd.get_dummies(data)
    data.drop('id',axis=1,inplace=True)
    data.fillna(0,inplace=True)
    return data
df_train = conv_cat(df_train)
df_test = conv_cat(df_test)

<div class="alert alert-block alert-warning"><em>Test Dataset

In [7]:
df_test

CustomerType  Age  TypeOfTravel  Class  FlightDistance  \
0                 1   52             1      0             160   
1                 1   36             1      2            2863   
2                 0   20             1      0             192   
3                 1   49             1      0            1182   
4                 1   16             1      0             311   
...             ...  ...           ...    ...             ...   
24327             0   34             1      2             526   
24328             1   23             1      2             646   
24329             1   17             0      0             828   
24330             1   14             1      2            1127   
24331             1   42             0      0             264   

       InflightWifiservice  DepartureArrivalTimeCconvenient  \
0                        5                                4   
1                        1                                1   
2                        2                                0   
3                        2                                3   
4                        3                                3   
...                    ...                              ...   
24327                    3                                3   
24328                    4                                4   
24329                    2                                5   
24330                    3                                3   
24331                    2                                5   

       EaseOfOnlineBooking  GateLocation  FoodAndDrink  ...  LegRoomService  \
0                        3             4             3  ...               5   
1                        3             1             5  ...               4   
2                        2             4             2  ...               1   
3                        4             3             4  ...               2   
4                        3             3             5  ...               3   
...                    ...           ...           ...  ...             ...   
24327                    3             1             4  ...               2   
24328                    4             4             4  ...               5   
24329                    1             5             2  ...               3   
24330                    3             3             4  ...               2   
24331                    2             5             4  ...               2   

       BaggageHandling  CheckinService  InflightService  Cleanliness  \
0                    5               2                5            5   
1                    4               3                4            5   
2                    3               2                2            2   
3                    2               4                2            4   
4                    1               1                2            5   
...                ...             ...              ...          ...   
24327                4               4                5            4   
24328                5               5                5            4   
24329                4               5                4            2   
24330                5               4                5            4   
24331                1               1                1            1   

       DepartureDelayInMinutes  ArrivalDelayInMinutes  satisfaction  \
0                           50                   44.0             1   
1                            0                    0.0             1   
2                            0                    0.0             0   
3                            0                   20.0             1   
4                            0                    0.0             1   
...                        ...                    ...           ...   
24327                        0                    0.0             0   
24328                        0                    0.0             1   
24329           

<div class="alert alert-block alert-warning"><em>Test Dataset

In [8]:
df_train

CustomerType  Age  TypeOfTravel  Class  FlightDistance  \
0                 1   13             0      1             460   
1                 0   25             1      2             235   
2                 1   26             1      2            1142   
3                 1   25             1      2             562   
4                 1   61             1      2             214   
...             ...  ...           ...    ...             ...   
97242             0   23             1      0             192   
97243             1   49             1      2            2347   
97244             0   30             1      2            1995   
97245             0   22             1      0            1000   
97246             1   27             1      2            1723   

       InflightWifiservice  DepartureArrivalTimeCconvenient  \
0                        3                                4   
1                        3                                2   
2                        2                                2   
3                        2                                5   
4                        3                                3   
...                    ...                              ...   
97242                    2                                1   
97243                    4                                4   
97244                    1                                1   
97245                    1                                1   
97246                    1                                3   

       EaseOfOnlineBooking  GateLocation  FoodAndDrink  ...  LegRoomService  \
0                        3             1             5  ...               3   
1                        3             3             1  ...               5   
2                        2             2             5  ...               3   
3                        5             5             2  ...               5   
4                        3             3             4  ...               4   
...                    ...           ...           ...  ...             ...   
97242                    2             3             2  ...               1   
97243                    4             4             2  ...               5   
97244                    1             3             4  ...               2   
97245                    1             5             1  ...               5   
97246                    3             3             1  ...               1   

       BaggageHandling  CheckinService  InflightService  Cleanliness  \
0                    4               4                5            5   
1                    3               1                4            1   
2                    4               4                4            5   
3                    3               1                4            2   
4                    4               3                3            3   
...                ...             ...              ...          ...   
97242                4               2                3            2   
97243                5               5                5            4   
97244                4               5                5            4   
97245                1               5                4            1   
97246                4               4                3            1   

       DepartureDelayInMinutes  ArrivalDelayInMinutes  satisfaction  \
0                           25                   18.0             0   
1                            1                    6.0             0   
2                            0                    0.0             1   
3                           11                    9.0             0   
4                            0                    0.0             1   
...                        ...                    ...           ...   
97242                        3                    0.0             0   
97243                        0                    0.0             1   
97244           

<div class="alert alert-block alert-info"><em>The 95% confidence t-interval for the mean number of Flight Distance</em></div>

In [9]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
from math import *


# df_train = pd.read_csv("Airline_Passenger_Satisfaction_Train_Cleaned.csv")
flight_distance = df_train['FlightDistance']

flight_distance_mean = flight_distance.mean()
flight_distance_std = flight_distance.std()
flight_distance_count = len(flight_distance)

print(flight_distance_mean)
print(flight_distance_std)
print(flight_distance_count)

t_value_1 = stats.t.ppf(0.975, 97246)
print(t_value_1)

conf_interval_from = flight_distance_mean - t_value_1 * (flight_distance_std / sqrt(flight_distance_count))
conf_interval_to = flight_distance_mean + t_value_1 * (flight_distance_std / sqrt(flight_distance_count))
print (conf_interval_from, conf_interval_to)
print(t_value_1 * (flight_distance_std / sqrt(flight_distance_count)))

1023.4863389101978
792.5745234030607
97247
1.9599883793763957
1018.5048920890965 1028.467785731299
4.98144682110124


<div class="alert alert-block alert-info"><em>The 95% confidence interval for the population mean Arrival Delay Minutes for those passengers who are both loyal customers with the 100 minutes of Departure Delay and are more than 50 in age.</em></div>

In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
from math import *

        
# df_train = pd.read_csv("Airline_Passenger_Satisfaction_Train_Cleaned.csv")

new_df = df_train.loc[(df_train["CustomerType"] == 1) & (df_train['DepartureDelayInMinutes'] == 100) & (df_train['Age'] > 50)]
arrival_delay_in_minutes = new_df['ArrivalDelayInMinutes']

arrival_delay_in_minutes_mean = arrival_delay_in_minutes.mean()
arrival_delay_in_minutes_std = arrival_delay_in_minutes.std()

arrival_delay_in_minutes_count = len(arrival_delay_in_minutes)

print(arrival_delay_in_minutes_mean)
print(arrival_delay_in_minutes_std)
print(arrival_delay_in_minutes_count)

t_value_2 = stats.t.ppf(0.975, 14)
print(t_value_2)

conf_interval_from = arrival_delay_in_minutes_mean - t_value_2 * (arrival_delay_in_minutes_std / sqrt(arrival_delay_in_minutes_count))
conf_interval_to = arrival_delay_in_minutes_mean + t_value_2 * (arrival_delay_in_minutes_std / sqrt(arrival_delay_in_minutes_count))
print (conf_interval_from, conf_interval_to)
print(t_value_2 * (arrival_delay_in_minutes_std / sqrt(arrival_delay_in_minutes_count)))

96.33333333333333
21.24572786187737
15
2.1447866879169273
84.56784140632003 108.09882526034663
11.765491927013306


In [13]:
df_train['FlightDistance'].describe()

count    97247.000000
mean      1023.486339
std        792.574523
min         31.000000
25%        401.000000
50%        762.000000
75%       1521.000000
max       3201.000000
Name: FlightDistance, dtype: float64

<div class="alert alert-block alert-info"><em>The Flight Distance of passengers is normally distributed, with mean 1023.486339 minutes, and standard deviation of 792.574523 minutes. For the following sample sizes, construct a 95% confidence interval for the population mean flight distance. Comment on the relationship between the confidence interval and the sample size.

a. n = 25

b. n = 100

c. n = 400


In [14]:
mean = 1023.486339
std = 792.574523

print('Sample Size = 25;')
t_value_3 = stats.t.ppf(0.975, 24)

print(t_value_3)

conf_interval_from = mean - t_value_3 * (std / sqrt(25))
conf_interval_to = mean + t_value_3 * (std / sqrt(25))
print (conf_interval_from, conf_interval_to)

#########################################################################

print('\nSample Size = 100;')
t_value_4 = stats.t.ppf(0.975, 99)

print(t_value_4)

conf_interval_from = mean - t_value_4 * (std / sqrt(100))
conf_interval_to = mean + t_value_4 * (std / sqrt(100))
print (conf_interval_from, conf_interval_to)

#########################################################################

print('\nSample Size = 400;')
t_value_5 = stats.t.ppf(0.975, 399)

print(t_value_5)

conf_interval_from = mean - t_value_5 * (std / sqrt(400))
conf_interval_to = mean + t_value_5 * (std / sqrt(400))
print (conf_interval_from, conf_interval_to)

Sample Size = 25;
2.0638985616280205
696.3276553994572 1350.645022600543

Sample Size = 100;
1.9842169515086827
866.2223586129492 1180.750319387051

Sample Size = 400;
1.965927295920809
945.5791445591443 1101.393533440856


<div class="alert alert-block alert-info"><em>The 95% confidence interval for the proportion π of the satisfaction among the
entire population of the airline’s customers

In [15]:
from scipy import stats
from math import *

cust_satisfaction = df_train['satisfaction']
cust_satisfaction_count = len(cust_satisfaction)
cust_satisfaction_rate = len(cust_satisfaction[cust_satisfaction == 1]) / len(cust_satisfaction)

z_value = stats.norm.ppf(0.975)
print(z_value)

conf_interval_from = cust_satisfaction_rate - z_value * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count)
conf_interval_to = cust_satisfaction_rate + z_value * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count)
print (conf_interval_from, conf_interval_to)

print(z_value * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count))

1.959963984540054
0.40694710604037987 0.4131296058376215
0.003091249898620817


<div class="alert alert-block alert-info"><em>Of 1000 customers who flew with the airline, 100 were satisfied. 
For the following confidence levels, construct a confidence interval for the population proportion
who would be satisfied of the flights. Comment on the relationship between the confidence interval width and the confidence levels.

a. 90%

b. 95%

c. 99%

In [16]:
total_customers = 1000
satisfied = 100
satisfied_rate = satisfied / total_customers

print('Confidence Interval: 90%')

z_value_1 = stats.norm.ppf(0.95)

print(z_value_1)

conf_interval_from = satisfied_rate - z_value_1 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
conf_interval_to = satisfied_rate + z_value_1 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
print (conf_interval_from, conf_interval_to)
print(z_value_1 * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count))

#########################################################################

print('\nConfidence Interval: 95%')

z_value_2 = stats.norm.ppf(0.975)

print(z_value_2)

conf_interval_from = satisfied_rate - z_value_2 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
conf_interval_to = satisfied_rate + z_value_2 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
print (conf_interval_from, conf_interval_to)
print(z_value_2 * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count))

#########################################################################

print('\nConfidence Interval: 99%')

z_value_3 = stats.norm.ppf(0.995)

print(z_value_3)

conf_interval_from = satisfied_rate - z_value_3 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
conf_interval_to = satisfied_rate + z_value_3 * sqrt((satisfied_rate * (1 - satisfied_rate))/total_customers)
print (conf_interval_from, conf_interval_to)
print(z_value_3 * sqrt((cust_satisfaction_rate * (1 - cust_satisfaction_rate))/cust_satisfaction_count))

Confidence Interval: 90%
1.6448536269514722
0.08439554836373328 0.11560445163626673
0.0025942586943775094

Confidence Interval: 95%
1.959963984540054
0.08140614903086316 0.11859385096913685
0.003091249898620817

Confidence Interval: 99%
2.5758293035489004
0.07556353761094071 0.1244364623890593
0.0040625910150734945


<div class="alert alert-block alert-info"><em>Testing whether mean Arrival Delay Minutes for those passengers who are both loyal customers with the 100 minutes of Departure Delay and are more than 50 in age differs from 2.4, with a level of significance α=0.05

- 𝐻0: 𝜇 = 2.4 𝑣s 𝐻𝑎: 𝜇 ≠ 2.4
- Null hypothesis is rejected if p-value < 0.05
- We also know that:
   - µ0 = 2.4 𝑥̅= 96.33
   - 𝑠 = 21.24 𝑛 = 15

In [17]:
tt = (96.33 - 2.4)/ (21.24/sqrt(15))
print(tt)

pval = stats.t.sf(np.abs(tt), 14)*2
print(pval)

17.127557707592402
8.697519812732835e-11


<div class="alert alert-block alert-warning"><em>p-value is greater than 0.05 which means that the null hypothesis is not rejected.

<div class="alert alert-block alert-info"><em>As 1142 of 97248 customer were satisfied, the estimate of the population proportion
π of all the customers that were satisfied is 0.01174.

Testing at α=0.10 whether π differs from 0.15, the hypotheses are:
𝐻0: 𝜋 = 0.15 𝑣s. 𝐻𝑎: 𝜋 ≠ 0.15

In [18]:
z_data = (0.01174 - 0.15)/ sqrt((0.15* 0.85)/97248)
print(z_data)

pval = stats.norm.sf(abs(z_data)) * 2
print (pval)

-120.74851527345388
0.0


<div class="alert alert-block alert-warning"><em>Here, z_data is less than 0 and p-value is 0.0 which means that we would reject H0. (p-value is less than α)